src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private int size = 0;
+    private T[] objects;
+
+    public ArrayList() {
+        objects = (T[]) new Object[10];
+    }
+
     @Override
     public void add(T value) {
-
+        increasingTheSize();
+        objects[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (size <= index || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index.");
+        }
+        increasingTheSize();
+        System.arraycopy(objects, index, objects, index + 1, size - index);
+        objects[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            increasingTheSize();
+            objects[size++] = list.get(i);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (size <= index || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index.");
+        }
```

Comment:
as described before, let's move to the separate method and reuse it everywhere. Also exception message isn't clear enough, let's add the index number